### PR TITLE
Expose credential off of JobTemplate

### PR DESCRIPTION
**Pull request by syncrou**
The credential attribute was simply returning the id vs
running a lookup and grabbing the AnsibleTowerClient::Credential object

### CODE REVIEW

**Comment by syncrou** [comment]
@miq-bot add_label bug

@miq-bot assign @bdunne

**Comment by syncrou** [comment]
@bdunne, @durandom - We weren't actually doing a lookup on credential as would be expected. I added the lookup in a similar pattern to how we grab the `survey_spec` and/or a `Job`

**Comment by syncrou** [comment]
@bdunne - If everyone is happy with this solution - I can do a followup with `AnsibleTowerClient::CloudCredential` and `AnsibleTowerClient::NetworkCredential`.

File: lib/ansible_tower_client/base_models/job_template.rb [hunk 1]
```diff
@@ -7,6 +7,12 @@ def launch(options = {})
       api.jobs.find(job['job'])
     end
 
+    def credential
+      cred_id = to_hash['credential']
+      return nil unless cred_id
+      api.credentials.find(cred_id)
```
Comment (gmcculloug) [inline lib/ansible_tower_client/base_models/job_template.rb]: Last two lines can be combined: `api.credentials.find(cred_id) if cred_id `

**Comment by durandom** [comment]
@syncrou in miq ansible refresh I am using the `_id` fields https://github.com/ManageIQ/manageiq/pull/13834/files#diff-46a11c7f5485160ff1f90f895e64e765R37 - so I dont need this magic

But while at it, maybe [the second part of this issue](https://github.com/ansible/ansible_tower_client_ruby/issues/68#issuecomment-278728495) is related?

**Pull request closed without merging**